EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
+        {
+            AddTextAsListItems(_contentProvider.Fonts[Constants.FontSize09],
+                insertLineBreaks: true,
+                new List<Action> { ShowLeaveGuildMessageBox },
+                _localizedStringFinder.GetString(EOResourceID.GUILD_LEAVE_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_YOU_ARE_ABOUT_TO_LEAVE_THE_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_REMEMBER_THAT_AFTER_YOU_HAVE_LEFT_THE_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_CLICK_HERE_TO_LEAVE_YOUR_GUILD)
+            );
         }
 
-        private void SetStateIfGuildMember(GuildDialogState state)
+        private void SetupRegisterGuildState()
+        {
+            AddTextAsListItems(_contentProvider.Fonts[Constants.FontSize09],
+                insertLineBreaks: true,
+                new List<Action> { ShowRegisterGuildMessageBox },
+                _localizedStringFinder.GetString(EOResourceID.GUILD_REGISTER_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_YOU_ARE_ABOUT_TO_CREATE_A_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_YOU_NEED_TO_HAVE_AT_LEAST_TEN_MEMBERS),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_THE_GUILD_MASTER_WILL_ASK_A_FEE),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_CLICK_HERE_TO_REGISTER_A_GUILD)
+            );
+        }
+
+        private void SetupWaitingForMembersState()
+        {
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 0)
+            {
+                PrimaryText = _localizedStringFinder.GetString(EOResourceID.GUILD_PLEASE_WAIT_FOR_ALL_MEMBERS_TO_JOIN)
+            }, false);
+
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 1), false);
+
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 0)
+            {
+                PrimaryText = Capitalize(_characterProvider.MainCharacter.Name)
+            }, false);
+        }
+
+        private void SetStateIfInGuild(State state)
         {
             if (!_characterProvider.MainCharacter.InGuild)
             {
-                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_NOT_IN_GUILD);
+                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_ALREADY_A_MEMBER);
                 dlg.ShowDialog();
                 return;
             }
 
             SetState(state);
         }
+
+        private void SetStateIfNotInGuild(State state)
+        {
+            if (_characterProvider.MainCharacter.InGuild)
+            {
+                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_ALREADY_A_MEMBER);
+                dlg.ShowDialog();
+                return;
+            }
+
+            SetState(state);
+        }
+
+        private void ShowJoinGuildMessageBox()
```

Comment:
Rename this to `JoinGuildClickAction` or something indicating it's a handler for clicking that link. Similar for any other click handlers.